### PR TITLE
grace join/spilling: try to free memory more often

### PR DESCRIPTION
**Pull request by yumkam**
number of rows per one loop may be large, try to free memory once in a while (0x10000 constant is somewhat arbitrary)

### Changelog entry 

...

### Changelog category 

* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp [hunk 1]
```diff
@@ -836,15 +836,19 @@ class TGraceJoinSpillingSupportState : public TComputationValue<TGraceJoinSpilli
...
     UpdateSpilling();
 
-    if (!HasMemoryForProcessing() && !IsSpillingFinalized) {
-        bool isWaitingForReduce = TryToReduceMemoryAndWait();
-        if (isWaitingForReduce) return;
-    }
-
+    ui64 cntrows = 0;
     while (*HaveMoreLeftRows || *HaveMoreRightRows) {
+        if (0 == (cntrows++ % CheckMemoryRowLimit)) { // checks before first loop and
```
Comment (lll-phill-lll) [inline ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp:841]: `cntrows` redundant. We can easily check memory on each row this call is as simple as checking the value of bool variable.

File: ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp [hunk 2]
```diff
@@ -836,15 +836,19 @@ class TGraceJoinSpillingSupportState : public TComputationValue<TGraceJoinSpilli
...
+                                                      // every CheckMemoryRowLimit rows
+            if (!HasMemoryForProcessing() && !IsSpillingFinalized) {
+                bool isWaitingForReduce = TryToReduceMemoryAndWait();
+                if (isWaitingForReduce) break;
+            }
+        }
         bool isYield = FetchAndPackData(ctx);
-        if (isYield) return;
+        if (isYield) break;
```
Comment (lll-phill-lll) [inline ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp:849]: What is the reason for this change?

**Review by lll-phill-lll — CHANGES_REQUESTED** [review]

File: ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp (hunk 1 above)
Comment (yumkam) [reply ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp]: Ok

File: ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp (hunk 2 above)
Comment (yumkam) [reply ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp]: It helps logging and should not affect result (but, indeed, this is unnecessary clutter)

**Review by lll-phill-lll — APPROVED** [review]
